tensorflow/compiler/tf2tensorrt/trt_convert_api.cc

```diff
@@ -63,8 +64,15 @@ Status RunGrappler(const MetaGraphDef& meta_graph_def,
         "Failed to create grappler item from MetaGraphDef.");
   }
 
+  std::unordered_map<string, tensorflow::DeviceProperties> device_map;
+  string device_name = "/job:localhost/replica:0/task:0/device:GPU:0";
```

Comment:
Why it is safe to assume GPU:0 is used?

Reply:
You are right that we should not have this as a constant. I have changed the cluster initialization, to mirror how the cluster is initialized in the python wrapper
https://github.com/tensorflow/tensorflow/blob/6cda859dbcc86772efc0588ede824ab47ac437f9/tensorflow/python/grappler/cluster_wrapper.cc#L82-L92 which is used when we initiate grappler conversion using the Python API https://github.com/tensorflow/tensorflow/blob/6cda859dbcc86772efc0588ede824ab47ac437f9/tensorflow/python/grappler/tf_optimizer.py#L63

This way both the C++ and the Python API of TF-TRT are calling Grappler with the same cluster settings.